source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,308 @@
+
+        auto u = xAxis / cv::norm(xAxis, cv::NORM_L2);
+        auto v = yAxis / cv::norm(yAxis, cv::NORM_L2);
+        auto normal = u.cross(v);
+        auto unitNormal = normal / cv::norm(normal, cv::NORM_L2);
+
+        auto rotMatrix = cv::Mat(3, 3, CV_64F);
+        for(int i = 0; i < 3; ++i)
+        {
+            rotMatrix.at<double>(i, 0) = u[i];
+            rotMatrix.at<double>(i, 1) = v[i];
+            rotMatrix.at<double>(i, 2) = unitNormal[i];
+        }
+
+        auto transform = transformationMatrix(rotMatrix, origin);
+
+        if(display)
+        {
+            displayMarkerPose(pointCloud, corners2D, centers2D);
+        }
+
+        return transform;
+    }
+
+    Zivid::Matrix4x4 invertAffineTransformation(const Zivid::Matrix4x4 &transform)
+    {
+        auto rot = cv::Mat(3, 3, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                rot.at<float>(i, j) = transform(i, j);
+            }
+        }
+
+        auto trans = cv::Mat(3, 1, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            trans.at<float>(i, 0) = transform(i, 3);
+        }
+
+        cv::Mat newRot = rot.inv();
+        cv::Mat newTrans = -newRot * trans;
+
+        auto cvInverted = cv::Mat(4, 4, CV_32F, 0.0);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                cvInverted.at<float>(i, j) = newRot.at<float>(i, j);
+            }
+        }
+        for(int i = 0; i < 3; ++i)
+        {
+            cvInverted.at<float>(i, 3) = newTrans.at<float>(i, 0);
+        }
+        cvInverted.at<float>(3, 3) = 1.0;
+
+        auto *ptr = &cvInverted.at<float>(0, 0);
+        auto invertTransform = Zivid::Matrix4x4(ptr, ptr + 16);
+
+        return invertTransform;
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        const auto dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ArucoSample.zdf";
+        std::cout << "Reading ZDF frame from file: " << dataFile << std::endl;
+        const Zivid::Frame frame = Zivid::Frame(dataFile);
+        auto pointCloud = frame.pointCloud();
+
+        std::cout << "Converting from RGB to OpenCV BGR image" << std::endl;
+        cv::Mat image = pointCloudToBGR(pointCloud);
+
+        std::cout << "Detecting aruco markers in 2D image" << std::endl;
+        auto markers = aruconano::MarkerDetector::detect(image);
+
+        std::cout << "Drawing marker corners" << std::endl;
+        for(const auto &marker : markers)
+        {
+            marker.draw(image);
+        }
+        std::cout << "Displaying detected markers" << std::endl;
+        cv::imshow("Detected Aruco markers", image);
+        cv::waitKey(0);
+
+        std::cout << "Detecting aruco marker poses" << std::endl;
+        const auto display = true;
+        auto transform = getMarkerPose(pointCloud, markers[0], display);
```

Comment:
why is this a variable? Why not use it only in line 291?

Reply:
no need, I removed the display option.